Fix vif unplug operations failing on RL9.3

priteau opened the pull request:
An instance on a hypervisor running Rocky Linux 9.3 would fail to resize or cold migrate with the error:

    Traceback (most recent call last):
      File "/var/lib/kolla/venv/lib/python3.9/site-packages/os_vif/__init__.py", line 110, in unplug
        plugin.unplug(vif, instance_info)
      File "/var/lib/kolla/venv/lib/python3.9/site-packages/vif_plug_ovs/ovs.py", line 445, in unplug
        self._unplug_bridge(vif, instance_info)
      File "/var/lib/kolla/venv/lib/python3.9/site-packages/vif_plug_ovs/ovs.py", line 376, in _unplug_bridge
        linux_net.delete_bridge(vif.bridge_name, v1_name)
      File "/var/lib/kolla/venv/lib/python3.9/site-packages/oslo_privsep/priv_context.py", line 271, in _wrap
        return self.channel.remote_call(name, args, kwargs,
      File "/var/lib/kolla/venv/lib/python3.9/site-packages/oslo_privsep/daemon.py", line 215, in remote_call
        raise exc_type(*result[2])
    pr2modules.netlink.exceptions.NetlinkError: (95, 'Operation not supported')

This appears to be caused by incompatibility between pyroute2<0.6.10 and the Rocky Linux 9.3 kernel. More details are available on Launchpad [1].

[1] https://bugs.launchpad.net/ubuntu/+source/pyroute2/+bug/1995469

priteau committed:
Fix vif unplug operations failing on RL9.3

An instance on a hypervisor running Rocky Linux 9.3 would fail to resize
or cold migrate with the error:

    Traceback (most recent call last):
      File "/var/lib/kolla/venv/lib/python3.9/site-packages/os_vif/__init__.py", line 110, in unplug
        plugin.unplug(vif, instance_info)
      File "/var/lib/kolla/venv/lib/python3.9/site-packages/vif_plug_ovs/ovs.py", line 445, in unplug
        self._unplug_bridge(vif, instance_info)
      File "/var/lib/kolla/venv/lib/python3.9/site-packages/vif_plug_ovs/ovs.py", line 376, in _unplug_bridge
        linux_net.delete_bridge(vif.bridge_name, v1_name)
      File "/var/lib/kolla/venv/lib/python3.9/site-packages/oslo_privsep/priv_context.py", line 271, in _wrap
        return self.channel.remote_call(name, args, kwargs,
      File "/var/lib/kolla/venv/lib/python3.9/site-packages/oslo_privsep/daemon.py", line 215, in remote_call
        raise exc_type(*result[2])
    pr2modules.netlink.exceptions.NetlinkError: (95, 'Operation not supported')

This appears to be caused by incompatibility between pyroute2<0.6.10 and
the Rocky Linux 9.3 kernel. More details are available on Launchpad [1].

[1] https://bugs.launchpad.net/ubuntu/+source/pyroute2/+bug/1995469

markgoddard reviewed (commented):
LGTM, will you update the nova images too?

seunghun1ee committed:
Fix no response from neutron metadata server during cloud-init

Instances created on a hypervisor running on Rocky Linux kernel 5.14.0-362.13.1.el9_3 fail to get response from metadata server due to pyroute2 incompatibilty.
Which led ssh key authentication failure.

priteau commented:
This is not needed because upstream is fixing it differently: https://review.opendev.org/c/openstack/kolla/+/904781
